llvm/lib/CodeGen/Analysis.cpp

```diff
@@ -739,6 +742,16 @@ bool llvm::returnTypeIsEligibleForTailCall(const Function *F,
   return true;
 }
 
+bool llvm::funcReturnsFirstArgOfCall(const CallInst &CI) {
+    const ReturnInst *Ret = dyn_cast<ReturnInst>(CI.getParent()->getTerminator());
+    Value *RetVal = Ret ? Ret->getReturnValue() : nullptr;
+    bool ReturnsFirstArg = false;
+    if (RetVal && ((RetVal == CI.getArgOperand(0) ||
+                    isPointerBitcastEqualTo(RetVal, CI.getArgOperand(0)))))
```

Comment:
I remember you mentioned the `ret ptr undef` case yesterday. Worth accounting for that here?